chains/ethereum/writer_test.go

```diff
@@ -166,7 +120,70 @@ func TestCreateAndExecuteErc20DepositProposal(t *testing.T) {
+
+	aliceConn, alice := createTestWriter(t, aliceTestConfig, contracts)
+	defer aliceConn.Close()
+	bobConn, bob := createTestWriter(t, bobTestConfig, contracts)
+	defer bobConn.Close()
+
+	// We'll use alice to setup the erc20
+	opts, nonce, err := aliceConn.newTransactOpts(big.NewInt(0), big.NewInt(DefaultGasLimit), big.NewInt(DefaultGasPrice))
+	nonce.lock.Unlock() // We manual increment nonce in tests
+	if err != nil {
+		t.Fatal(err)
+	}
+	erc20Address := ethtest.DeployMintApproveErc20(t, aliceConn.conn, opts, contracts.ERC20HandlerAddress, big.NewInt(100))
+	ethtest.FundErc20Handler(t, aliceConn.conn, opts, contracts.ERC20HandlerAddress, erc20Address, big.NewInt(100))
 
+	// Create initial transfer message
+	resourceId := msg.ResourceIdFromSlice(append(common.LeftPadBytes(erc20Address.Bytes(), 31), 0))
+	recipient := ethcrypto.PubkeyToAddress(bob.conn.kp.PrivateKey().PublicKey).Bytes()
+	amount := big.NewInt(10)
+	m := msg.NewFungibleTransfer(1, 0, 0, amount, resourceId, recipient)
+	ethtest.RegisterErc20Resource(t, aliceConn.conn, opts, contracts.ERC20HandlerAddress, resourceId, erc20Address)
+	// Helpful for debugging
+	go watchEvent(alice.conn, utils.DepositProposalCreated)
+	go watchEvent(alice.conn, utils.DepositProposalVote)
+	go watchEvent(alice.conn, utils.DepositProposalFinalized)
+	go watchEvent(alice.conn, utils.DepositProposalExecuted)
+
+	routeMessageAndWait(t, alice, bob, m)
+}
+
+func TestCreateAndExecuteErc721Proposal(t *testing.T) {
+	contracts := deployTestContracts(t, aliceTestConfig.id)
+
+	aliceConn, alice := createTestWriter(t, aliceTestConfig, contracts)
+	defer aliceConn.Close()
+	bobConn, bob := createTestWriter(t, bobTestConfig, contracts)
+	defer bobConn.Close()
+
+	// We'll use alice to setup the erc721
+	tokenId := big.NewInt(1)
+	opts, nonce, err := aliceConn.newTransactOpts(big.NewInt(0), big.NewInt(DefaultGasLimit), big.NewInt(DefaultGasPrice))
+	nonce.lock.Unlock() // We manual increment nonce in tests
+	if err != nil {
+		t.Fatal(err)
+	}
+	erc721Address := ethtest.DeployMintApproveErc721(t, aliceConn.conn, opts, contracts.ERC721HandlerAddress, tokenId)
+	ethtest.FundErc721Handler(t, aliceConn.conn, opts, contracts.ERC721HandlerAddress, erc721Address, tokenId)
+
+	// Create initial transfer message
+	resourceId := msg.ResourceIdFromSlice(append(common.LeftPadBytes(erc721Address.Bytes(), 31), 0))
+	recipient := ethcrypto.PubkeyToAddress(bob.conn.kp.PrivateKey().PublicKey).Bytes()
+	m := msg.NewNonFungibleTransfer(1, 0, 0, resourceId, tokenId, recipient, []byte{})
+	ethtest.RegisterErc721Resource(t, aliceConn.conn, opts, contracts.ERC721HandlerAddress, resourceId, erc721Address)
+	// Helpful for debugging
+	go watchEvent(alice.conn, utils.DepositProposalCreated)
+	go watchEvent(alice.conn, utils.DepositProposalVote)
+	go watchEvent(alice.conn, utils.DepositProposalFinalized)
+	go watchEvent(alice.conn, utils.DepositProposalExecuted)
+
+	routeMessageAndWait(t, alice, bob, m)
```

Comment:
This doesn't verify execute deposit operated correctly, it just ensures that the method did not fail. Is this intended?

Reply:
Added